android/build.gradle

```diff
@@ -193,8 +194,75 @@ subprojects { subproject ->
         if ((subproject.plugins.hasPlugin('android')
                     || subproject.plugins.hasPlugin('android-library'))
                 && rootProject.ext.has('buildToolsVersion')) {
+
             android {
                 buildToolsVersion rootProject.ext.buildToolsVersion
+
+                // Automatically detect namespace from Java package structure
+                def javaSrcDir = new File(subproject.projectDir, 'src/main/java')
+                def detectedNamespace = null
+
+                if (javaSrcDir.exists()) {
+                    def javaFiles = fileTree(javaSrcDir).include('**/*.java')
+                    if (!javaFiles.isEmpty()) {
+                        def firstJavaFile = javaFiles.first()
+                        def relativePath = firstJavaFile.path.substring(javaSrcDir.path.length() + 1)
+                        def packagePath = relativePath.substring(0, relativePath.lastIndexOf('/'))
+                        detectedNamespace = packagePath.replace('/', '.')
+                    }
+                }
+
+                // Set namespace if not already set and we detected one
+                if (!android.hasProperty('namespace') || android.namespace == null) {
+                    if (detectedNamespace != null) {
+                        android.namespace = detectedNamespace
+                    } else if (project.group != null && project.group != '') {
+                        android.namespace = project.group
+                    } else {
+                        android.namespace = "com.${subproject.name.replaceAll('-', '_')}"
+                    }
+                }
+
+                // Only enable buildConfig if namespace is set
+                buildFeatures {
+                    if (android.namespace != null) {
+                        buildConfig true
+                    }
+                }
+
+                // Set JVM target across all subprojects
+                compileOptions {
+                    sourceCompatibility JavaVersion.VERSION_17
```

Comment:
I wonder if we can put this in the root project's ext, so we do something like:

```
compileOptions {
                    sourceCompatibility rootProject.ext.javaVersion
                    targetCompatibility rootProject.ext.javaVersion
}
```

This would make it easier to change.